src/NodeCompiler/CompileNodeToValue.php

```diff
@@ -47,6 +47,10 @@ public function __invoke(Node $node, CompilerContext $context)
             return $this->compileBinaryOperator($node, $context);
         }
 
+        if ($node instanceof Node\Scalar\MagicConst\Dir) {
+            return '__DIR__';
```

Comment:
Hmm, not sure this is correct; surely it is the directory in which the context's class sits in, not just the string literal `__DIR__`?